app/commands/exercise/authorship/create.rb

```diff
@@ -0,0 +1,21 @@
+class Exercise
+  class Authorship
+    class Create
+      include Mandate
+
+      initialize_with :exercise, :author
+
+      def call
+        authorship = exercise.authorships.create!(author: author)
+        User::ReputationAcquisition.find_or_create_by!(
+          user: author,
+          reason_object: authorship,
+          reason: "exercise_authorship",
+          category: "exercise_authorship"
+        )
```

Comment:
I was wondering if it would make sense to have strongly-typed commands here too. Something like:

```suggestion
        User::ReputationAcquisition::ExerciseAuthorship::Create.(author, authorship)
```

Or would that be overkill?

Reply:
"strongly-typed commands" - I like that phrase :)

And yes, I'd be pro-that.

Reply:
I've pushed a commit that implements the above. Let me know what you think